src/main/resources/reference.conf

```diff
@@ -1,41 +1,127 @@
 # ===================
 # Redis cache configuration
 # ===================
-play.cache.redis {
-  # redis server: host
-  host:       localhost
-  # redis server: port
-  port:       6379
-  # redis server: database number
-  database:   1
-  # list of nodes in a cluster
-  cluster:    [
-    # nodes are defined as a sequence of objects:
-    #  required string, defining a host the node is running on
-    #  host:        localhost
-    #  required integer, defining a port the node is running on
-    #  port:        6379
-    #  optional string, defines a password to use
-    #  password:    null
-  ]
-  # connection timeout in milliseconds
-  timeout:    1s
+play.cache.redis: {
+
+  # list of named caches available in the application
+  #
+  # for more details see https://www.playframework.com/documentation/2.6.x/ScalaCache#Accessing-different-caches
+  #
+  instances {
+
+    # definition of the default cache,
+    # its name is defined in 'play.cache.default-cache'
+    #
+    # when there should be more caches, defined them all under the 'instances' key.
+    #
+    # for backward compatibility, the 'host', 'port', etc.
+    # can be defined directly under 'play.cache.redis' key.
+    # However, in such a case, it overrides the configuration of the default
+    # cache and is reported as a warning because this configuration
+    # is considered deprecated since 2.0.0
+    #
+    play {
+
+      ##########################
+      # Standalone mode
+      ##########################
+
+      # redis server: host
+      host:       localhost
+      # redis server: port
+      port:       6379
+      # redis server: database number
+      database:   1
```

Comment:
Honestly, I don't mind that much for backward compatibility right now as there is none right now. The change is too big to keep it compatible I think.

But on the other hand, the complexity of the new configuration is worrying me, as I posted couple secs ago.